### PR TITLE
Add port attibute to pause and unpause the egress of a port

### PR DESCRIPTION
**Pull request by wendani**
Signed-off-by: Wenda Ni <wenni@microsoft.com>

### CODE REVIEW

File: inc/saiport.h [hunk 1]
```diff
@@ -1086,6 +1086,17 @@ typedef enum _sai_port_attr_t
...
+    /**
+     * @brief Pause port egress
+     *
+     * Enable/Disable egress pause of a port.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_PORT_ATTR_EGRESS_PAUSE_ENABLE,
```
Comment (itaibaz) [inline inc/saiport.h]: I think we can achieve the same functionality by
SAI_PORT_ATTR_QOS_SCHEDULER_PROFILE_ID
And using a profile with SAI_SCHEDULER_ATTR_MAX_BANDWIDTH_RATE of 0
Is this different ?

Comment (wendani) [reply inc/saiport.h]: This STOP_PORT_MAX_RATE approach does not work on brcm platform.

Comment (itaibaz) [reply inc/saiport.h]: Perhaps this should be a vendor specific attribute then ?
The scheduler profile approach is something all vendors agreed upon long time ago, having 2 ways to control the same functionality isn't the best approach IMO

Comment (lguohan) [reply inc/saiport.h]: scheduler profile approach is a mean rather than the purpose, usually SAI attribute defines the purpose not the means.

Comment (tushar-ty) [reply inc/saiport.h]: Request name change to TX_ENABLE or STOP_TX.
Comments should specify that this should not drop frames, just stop transmit and buffer.

**Comment by wendani** [comment]
retest this please

**Comment by wendani** [comment]
retest this please

**Comment by wendani** [comment]
retest this please

**Comment by wendani** [comment]
retest this please.

**Comment by lguohan** [comment]
please resolve build failure

**Comment by wendani** [comment]
--- is not allowed in the comment

**Pull request merged**